### DISCUSSION TITLE
ndk issue  in client build

### DISCUSSION THREAD

**Discussion by AlothNaveen** [strongswan/strongswan#2317] [Q&A] [answered]
export ANDROID_NDK_ROOT=/home/naveen/Ipsec_Project/strongswan/ndk
naveen@m3682029:~/strongswan/src/frontends/android/openssl$ export OPENSSL_SRC=~/openssl-1.1.1f
naveen@m3682029:~/strongswan/src/frontends/android/openssl$ export NO_DOCKER=true
naveen@m3682029:~/strongswan/src/frontends/android/openssl$ ./build.sh
~/openssl-1.1.1f ~/strongswan/src/frontends/android/openssl
## Building OpenSSL's libcrypto for arm64-v8a
make: *** No rule to make target 'clean'.  Stop.
Configuring OpenSSL version 1.1.1f (0x1010106fL) for android-arm64
Using os-specific seed configuration
$ANDROID_NDK_HOME=/home/naveen/Ipsec_Project/strongswan/ndk is invalid at (eval 10) line 37.

**Comment by tobiasbrunner** [comment, marked as answer]
As indicated by the error message, the path you passed as `ANDROID_NDK_ROOT` might be invalid for OpenSSL (i.e. might not contain an NDK or one that's incompatible with the version of OpenSSL you are using). In OpenSSL 1.1.1f, the script expects either a `platforms` directory or an `AndroidVersion.txt` file in the passed NDK directory. Note that this check has changed in later versions, so you might want to try a newer OpenSSL release.

**Reply by AlothNaveen** [reply]
can i use  [openssl-3.3.1.tar.gz](https://www.openssl.org/source/openssl-3.3.1.tar.gz) ??

**Reply by tobiasbrunner** [reply]
No, unless you patch the sources yourself, see [this OpenSSL ticket](https://github.com/openssl/openssl/pull/22181).

**Reply by tobiasbrunner** [reply]
I can't help you with that, sorry.

**Reply by tobiasbrunner** [reply]
I've no idea whatsoever.
